src/Duel.java

```diff
@@ -0,0 +1,154 @@
+public class Duel {
+    private static boolean turn = true;
+
+    public static void startDuel(Person gamer, Person enemy, Room room) {
+        while (gamer.getCurrentHealth() > 0 && enemy.getCurrentHealth() > 0) {
+            if (turn) {
+                Menu.frame();
+                Menu.gamerHealth(gamer);
+                Menu.enemyHealth(enemy);
+                Menu.gamerTurn();
+                gamerTurn(gamer, enemy, room);
+            } else {
+                Menu.frame();
+                Menu.enemyTurn();
+                enemyTurn(enemy, gamer, room);
+            }
+            if (enemy.getCurrentHealth() <= 0) {
+                Menu.enemyDeath();
+            }
+            if (gamer.getCurrentHealth() <= 0) {
+                Menu.deathMessage();
+            }
+        }
+        turn = true;
+    }
+
+    private static void gamerTurn(Person gamer, Person enemy, Room room) {
+        String type = gamer.getWeapon().getClass().getName();
+        switch (Menu.duelInteraction(gamer, enemy)) {
+            case 1 -> {
+                MeleeAttack(type, gamer.getWeapon(), enemy, room);
+                turn = false;
+            }
+            case 2 -> {
+                MeleeThrow(type, gamer.getWeapon(), enemy, room);
+                turn = false;
+            }
+            case 3 -> {
+                boolean success;
+                success = RangeAttack(type, gamer.getWeapon(), gamer, enemy, room);
+                if (success) {
+                    turn = false;
+                } else {
+                    Menu.needReload();
+                }
+            }
+            case 4 -> {
+                RangeReload(type, gamer.getWeapon());
+                turn = RangeReload(type, gamer.getWeapon());
+            }
+            case 5 -> {
+                enemy.setCurrentHealth(0);
+                int backDamage = 100 + (int) (Math.random() * 300);
+                gamer.setCurrentHealth(gamer.getCurrentHealth() - backDamage);
+                Menu.backShootMessage(backDamage);
+            }
+        }
+    }
+
+    private static void enemyTurn(Person gamer, Person enemy, Room room) {
+        String type = gamer.getWeapon().getClass().getName();
+        boolean success;
+        if (type.equals("MeleeWeapons")) {
+            MeleeAttack(type, gamer.getWeapon(), enemy, room);
+            turn = true;
+        } else if (type.equals("RangeWeapons")) {
+            success = RangeAttack(type, gamer.getWeapon(), gamer, enemy, room);
+            if (!success) {
```

Comment:
there was if else duplicate, move it to separate function and also replace constructor calling